src/snowflake/snowpark/mock/_stored_procedure.py

```diff
@@ -262,37 +262,29 @@ def _do_register_sp(
         _emit_ast: bool = True,
         **kwargs,
     ) -> StoredProcedure:
-
-        check_imports_type(imports, "stored-proc-level")
-
-        ast = None
-        stmt = None
-        if _emit_ast:
-            stmt = self._session._ast_batch.assign()
-            ast = with_src_position(stmt.expr.stored_procedure, stmt)
-            build_sproc(
-                ast,
+        if kwargs.get("_registered_object_name") is not None:
+            ast, ast_id = None, None
+            if _emit_ast:
+                stmt = self._session._ast_batch.assign()
+                ast = with_src_position(stmt.expr.stored_procedure, stmt)
+                ast_id = stmt.var_id.bitfield1
+
+            object_name = kwargs["_registered_object_name"]
+            sproc = MockStoredProcedure(
                 func,
                 return_type,
                 input_types,
-                sp_name,
-                stage_location,
-                imports,
-                packages,
-                replace,
-                if_not_exists,
-                parallel,
-                strict,
-                external_access_integrations,
-                secrets,
-                comment,
+                object_name,
+                Set(),
                 execute_as=execute_as,
-                statement_params=statement_params,
-                source_code_display=source_code_display,
-                is_permanent=is_permanent,
-                session=self._session,
-                **kwargs,
+                strict=strict,
+                _ast=ast,
+                _ast_id=ast_id,
             )
+            self._registry[object_name] = sproc
+            return sproc
+
+        check_imports_type(imports, "stored-proc-level")
```

Comment:
The check should come first, there were a few tests that break otherwise.

Reply:
I don't think so.  The check_imports_type should only occur on the client side since that's where the sproc is actually created.  On the TCM / execution playback side we expect the sproc object already exists (the branch above) and therefore any related validations have occurred.  Note the branch above is used by the no-execution backend for testing.  Let me know if there's specific tests you think would fail as not seeing them.